Wikipedia/Code/InsertMediaSearchResultsCollectionViewController.swift

```diff
@@ -0,0 +1,244 @@
+import UIKit
+import SafariServices
+
+fileprivate class FlowLayout: UICollectionViewFlowLayout {
+    private var oldBoundsWidth: CGFloat = 0
+
+    override init() {
+        super.init()
+        minimumInteritemSpacing = 8
+        minimumLineSpacing = 32
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func shouldInvalidateLayout(forBoundsChange newBounds: CGRect) -> Bool {
+        defer {
+            oldBoundsWidth = newBounds.width
+        }
+        return super.shouldInvalidateLayout(forBoundsChange: newBounds) || newBounds.width != oldBoundsWidth
+    }
+
+    override func invalidateLayout(with context: UICollectionViewLayoutInvalidationContext) {
+        defer {
+            super.invalidateLayout(with: context)
+        }
+
+        guard let collectionView = collectionView else {
+            return
+        }
+        sectionInset.left = collectionView.scrollIndicatorInsets.left
+        sectionInset.right = collectionView.scrollIndicatorInsets.right
+        let dimension = (collectionView.bounds.width / 3) - minimumInteritemSpacing * 2 - sectionInset.left - sectionInset.right
```

Comment:
The order of operations is a bit off here because all the spacing should be subtracted before dividing by the number of columns. I was also wrong about using `scrollIndicatorInsets` - it looks like the right calculation is:
```
let countOfColumns: CGFloat = 3
        sectionInset = UIEdgeInsets(top: 0, left: collectionView.layoutMargins.left, bottom: 0, right: collectionView.layoutMargins.right)
        let availableWidth = collectionView.bounds.width - minimumInteritemSpacing * (countOfColumns - 1) - collectionView.contentInset.left - collectionView.contentInset.right - sectionInset.left - sectionInset.right
        let dimension = floor(availableWidth / countOfColumns)
```

^ even with that the visible cells are the wrong size on rotate (but correct after scrolling them off and back on the screen), so may need some other invalidation or `reloadData` on bounds change in the VC